File: app-core/src/test/java/kr/flab/wiki/core/login/utils/UserTestUtils.java
package kr.flab.wiki.core.login.utils;

import com.github.javafaker.Faker;
import kr.flab.wiki.core.login.persistence.User;
import kr.flab.wiki.core.login.persistence.UserEntity;

public class UserTestUtils {
    public static User createRandomUserEntity() {
        Faker faker = StringRandomUtils.getFaker();
        return new UserEntity(
                faker.internet().emailAddress(),
                faker.internet().password()
        );
    }

    public static User createRandomUserEntity(UserType userType) {
        Faker faker = StringRandomUtils.getFaker();
        if(userType == UserType.BLANK_EMAIL){
            return new UserEntity("", faker.internet().password());
        }
        if(userType == UserType.BLANK_PASSWORD){
            return new UserEntity(faker.internet().emailAddress(), "");
        }
        if(userType == UserType.NOT_EMAIL){
            return new UserEntity(faker.animal().name(), faker.internet().password());
        }
        return new UserEntity("", "");
    }

    public enum UserType {
        BLANK_EMAIL,
        BLANK_PASSWORD,
        NOT_EMAIL
    }


}

File: app-core/src/test/kotlin/kr/flab/wiki/core/login/utils/UserTestUtils.kt
package kr.flab.wiki.core.login

import com.github.javafaker.Faker
import kr.flab.wiki.core.login.persistence.UserEntity
import java.util.*

object UserTestUtils {
    val faker : Faker = Faker(Locale.ENGLISH)
}

fun createRandomUserEntity() = UserEntity().apply {
    email = UserTestUtils.faker.internet().emailAddress()
    password = UserTestUtils.faker.internet().password()
}

fun createRandomUserEntity(type : UserType) =
    when(type){
        UserType.BLANK_EMAIL -> UserEntity().apply {
            email = ""
            password = UserTestUtils.faker.internet().password()
        }
        UserType.BLANK_PASSWORD -> UserEntity().apply {
            email = UserTestUtils.faker.internet().emailAddress()
            password = ""
        }
        UserType.NOT_EMAIL -> UserEntity().apply {
            email = UserTestUtils.faker.animal().name()
            password = ""
        }
    }

enum class UserType() {
    BLANK_EMAIL,
    BLANK_PASSWORD,
    NOT_EMAIL
}